libsovtoken/tests/payment_chaos_tests.rs

```diff
@@ -0,0 +1,357 @@
+        wallet.handle,
+        Some(&dids[0]),
+        &mint_output_json,
+        None,
+    ).unwrap();
+
+    trace!("{:?}", &mint_req);
+
+    let mint_req = Request::<MintRequest>::multi_sign_request(
+        wallet.handle,
+        &mint_req,
+        dids.clone()
+    ).unwrap();
+
+    trace!("{:?}", &mint_req);
+
+    let _ = indy::ledger::Ledger::submit_request(pool_handle, &mint_req).unwrap();
+}
+
+// ------------------------------------------------------------------------------------------------
+#[test]
+pub fn pay_without_outputs_fails() {
+
+    // ---- setup
+    sovtoken::api::sovtoken_init();
+    let wallet = Wallet::new();
+    let setup = Setup::new(&wallet, SetupConfig {
+        num_addresses: 1,
+        num_trustees: 4,
+        num_users: 0,
+        mint_tokens: None,
+        fees: None,
+    });
+    let payment_addresses = &setup.addresses;
+    let pool_handle = setup.pool_handle;
+    let dids = setup.trustees.dids();
+    let tokens : i32 = 50;
+
+    // ---- create some tokens
+    do_minting(pool_handle, &wallet, &dids, &payment_addresses[0], tokens);
+
+    // ---- create payment request with empty output json
+    let txo_1 = utils::payment::get_utxo::get_first_utxo_txo_for_payment_address(&wallet, pool_handle, dids[0], &payment_addresses[0]);
+
+    let pay_input_json = json!([
+        txo_1
+    ]).to_string();
+
+    let pay_output_json = json!([
+    ]).to_string();
+
+    match indy::payments::Payment::build_payment_req(wallet.handle, Some(dids[0]), &pay_input_json, &pay_output_json, None) {
+        Ok(_) => { assert!(false, "Expected CommonInvalidStructure error"); }
+        Err(ec) => { assert_eq!(ec, ErrorCode::CommonInvalidStructure); }
+    }
+
+}
+
+// ------------------------------------------------------------------------------------------------
+#[test]
+pub fn pay_without_inputs_fails() {
+
+    // ---- setup
+    sovtoken::api::sovtoken_init();
+    let wallet = Wallet::new();
+    let setup = Setup::new(&wallet, SetupConfig {
+        num_addresses: 1,
+        num_trustees: 4,
+        num_users: 0,
+        mint_tokens: None,
+        fees: None,
+    });
+    let payment_addresses = &setup.addresses;
+    let pool_handle = setup.pool_handle;
+    let dids = setup.trustees.dids();
+    let tokens : i32 = 50;
+
+    // ---- create some tokens
+    do_minting(pool_handle, &wallet, &dids, &payment_addresses[0], tokens);
+
+    // ---- create payment request with empty input json
+    let pay_input_json = json!([
+
+    ]).to_string();
+
+    let pay_output_json = json!([
+        {
+            "recipient": payment_addresses[0],
+            "amount": 50
+        }
+    ]).to_string();
+
+    match indy::payments::Payment::build_payment_req(wallet.handle, Some(dids[0]), &pay_input_json, &pay_output_json, None) {
+        Ok(_) => { assert!(false, "Expected CommonInvalidStructure error"); }
+        Err(ec) => { assert_eq!(ec, ErrorCode::CommonInvalidStructure); }
```

Comment:
You could use an assert_eq! assertion which is a little more condensed.

```
let ec = (...).unwrap_err();
assert_eq!(ErrorCode::CommonInvalidStructure, ec);
```
or
```
assert_eq!(Err(ErrorCode::CommonInvalidStructure), result);
```

Reply:
much better.  thnx